app/src/components/Modals/delete-city-modal.tsx

```diff
@@ -71,7 +71,7 @@ const DeleteCityDialog: FC<DeleteCityDialogProps> = ({
 
   return (
     <>
-      <DialogRoot preventScroll open={isOpen} onExitComplete={onClose}>
+      <DialogRoot preventScroll open={isOpen} onOpenChange={onClose}>
```

Comment:
Let's check if it's actually being checked by using the argument to determine the opening/ closing state (same for the other onOpenChange params below)